src/MSIdentityScaffolding/Microsoft.DotNet.MSIdentity/MicrosoftIdentityPlatform/MicrosoftIdentityPlatformApplicationManager.cs

```diff
@@ -192,44 +197,55 @@ internal async Task<bool> UpdateApplication(TokenCredential tokenCredential, App
                 //update redirect uris
                 List<string> existingRedirectUris = updatedApp.Web.RedirectUris.ToList();
                 List<string> urisToEnsure = ValidateUris(toolOptions.RedirectUris).ToList();
-                existingRedirectUris.AddRange(urisToEnsure);
-                updatedApp.Web.RedirectUris = existingRedirectUris.Distinct();
-
-                //update implicit grant settings
+                List<string> urisToAdd = existingRedirectUris.Except(urisToEnsure).ToList();
+                if (urisToAdd.Any())
+                {
+                    needsUpdate = true;
+                    existingRedirectUris.AddRange(urisToAdd);
+                    updatedApp.Web.RedirectUris = existingRedirectUris.Distinct();
```

Comment:
int originalUrisCount = existingRedirectUris.Count;
..... // update existingRedirectUris
if (updatedApp.Web.RedirectUris.Count > existingRedirectUris.Count)
{
    needsUpdate = true;
}

Reply:
hmm slight edit to it.

existingRedirectUris.AddRange(urisToEnsure);
List<string> updatedUriList = existingRedirectUris.Distinct().ToList();
if (updatedUriList.Count() > updatedApp.Web.RedirectUris.Count())
{
    needsUpdate = true;
}

makes more sense to me where the the new uri list count should be greater than the existing uri count unlike your eg above

Reply:
I think you meant 
if (updatedApp.Web.RedirectUris.Count > originalUrisCount)
looks good other than that.